### PR TITLE
DAOS-3730 tests: Starting agent and server with common params

### PR DESCRIPTION
**Pull request by phender**
Expanding upon the recent changes to the functional tests starting of
the daos_agent and daos_server in which yaml configuration files are
utilized, the common parts of those yaml files are now defined in a
single object.  This includes the transport credentials and access
points In addition a single class maintains the lists of all the agents
and servers being run in the test and it provides a single source from
which to obtain data required by the test from the agent or server
configuration files.

Signed-off-by: Phillip Henderson <phillip.henderson@intel.com>

### CODE REVIEW

File: src/tests/ftest/util/agent_utils.py [hunk 1]
```diff
@@ -20,279 +20,158 @@
...
 
-class AgentFailed(Exception):
-    """Agent didn't start/stop properly."""
-
-class DaosAgentConfig(ObjectWithParameters):
-    """Defines the daos_agent configuration yaml parameters."""
+from command_utils import BasicParameter, FormattedParameter, YamlParameters
+from command_utils import YamlCommand, SubprocessManager, CommandFailure
+from command_utils import TransportCredentials
+from general_utils import pcmd
```
[the remark this answers is not in the document]
Comment (phender) [reply src/tests/ftest/util/agent_utils.py]: Fixed

File: src/tests/ftest/util/agent_utils.py [hunk 2]
```diff
@@ -20,279 +20,158 @@
...
+            messages.append(
+                "Error stopping the orterun subprocess: {}".format(error))
+
+        # Kill any leftover processes that may not have been stopped correctly
+        self.kill()
+
+        # Report any errors after all stop actions have been attempted
+        if len(messages) > 0:
+            raise CommandFailure(
+                "Failed to stop agents:\n  {}".format("\n  ".join(messages)))
```
[the remark this answers is not in the document]
Comment (phender) [reply src/tests/ftest/util/agent_utils.py]: Fixed

File: src/tests/ftest/util/agent_utils.py [hunk 3]
```diff
@@ -20,279 +20,158 @@
...
   portions thereof marked with this legend must also reproduce the markings.
-'''
-from __future__ import print_function
-
-import os
-import time
-import subprocess
-import getpass
+"""
+from logging import getLogger
```
[the remark this answers is not in the document]
Comment (phender) [reply src/tests/ftest/util/agent_utils.py]: Fixed

File: src/tests/ftest/util/command_utils.py [hunk 4]
```diff
@@ -24,9 +24,13 @@
 from logging import getLogger
 import time
 import os
+import re
 import signal
+import yaml
 
 from avocado.utils import process
+from general_utils import pcmd, check_file_exists, stop_processes
```
[the remark this answers is not in the document]
Comment (phender) [reply src/tests/ftest/util/command_utils.py]: Fixed

File: src/tests/ftest/control/dmg_nvme_scan_test.py [hunk 5]
```diff
@@ -21,43 +21,24 @@
...
 """
-from __future__ import print_function
+from control_test_base import ControlTestBase
 
-import os
 
-from dmg_utils import DmgCommand
-from apricot import TestWithServers
-from avocado.utils import process
+class DmgNvmeScanTest(ControlTestBase):
```
[the remark this answers is not in the document]
Comment (phender) [reply src/tests/ftest/control/dmg_nvme_scan_test.py]: Fixed

File: src/tests/ftest/pool/info_tests.py [hunk 6]
```diff
@@ -51,7 +51,7 @@ def test_pool_info_query(self):
         self.pool.get_params(self)
         permissions = self.params.get("permissions", "/run/test/*")
         targets = self.params.get("targets", "/run/server_config/*")
-        pool_targets = len(self.hostlist_servers) * targets
+        # pool_targets = len(self.hostlist_servers) * targets
```
Comment (saurabhtandan) [inline src/tests/ftest/pool/info_tests.py]: Why is this being commented?

File: src/tests/ftest/util/command_utils.py [hunk 7]
```diff
@@ -289,7 +347,7 @@ def _run_process(self):
             "allow_output_check": "combined",
             "shell": True,
             "env": self.env,
-            "sudo": self.sudo,
+            # "sudo": self.sudo,
```
Comment (saurabhtandan) [inline src/tests/ftest/util/command_utils.py]: same as above

File: src/tests/ftest/util/dfuse_utils.py [hunk 8]
```diff
@@ -23,7 +23,7 @@
 from command_utils import ExecutableCommand, EnvironmentVariables
 from command_utils import CommandFailure, FormattedParameter
 from ClusterShell.NodeSet import NodeSet
-from server_utils import AVOCADO_FILE
+# from server_utils import AVOCADO_FILE
```
Comment (saurabhtandan) [inline src/tests/ftest/util/dfuse_utils.py]: same as above

File: src/tests/ftest/util/logger_utils.py [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+"""
+
+
+class TestLogger(object):
+    """Defines a Logger that that also logs messages to DaosLog."""
```
Comment (saurabhtandan) [inline src/tests/ftest/util/logger_utils.py]: typo : "that that"

Comment (phender) [reply src/tests/ftest/util/logger_utils.py]: Fixed - thanks!

**Pull request closed without merging**